From 45740bb9666f3db9b3a5a99c0c2e9847ca15370b Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Mon, 2 Apr 2018 20:55:20 +0800
Subject: [PATCH 3566/5242] MLK-17907-1 soc: imx8: Add compatible to let domu
 know the errata

commit  7448977b4de3b0ed3be88494985caa80a513f6e9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Because of the TO1.0 issue on i.MX8QM, we need to let DomU
know that errata, then DomU big.Little guest could work correctly.

This is not that good to add compatible for xen vm machine in
imx8 soc code.  Another method is to modify xen toolstack to create
the imx8qm compatible in domu dts, but this is not welcomed, from
xen maitainer's view "Domu should be not binded with a dedicated SoC".

For supporting i.MX8QM, Let's first use the compatible. This might
cause qxp Domu thinking it is 8QM, but we do not support passthrough
on 8QXP, because no smmu.

So it is ok to add the compatible here.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/soc-imx8.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/soc/imx/soc-imx8.c b/drivers/soc/imx/soc-imx8.c
index 65b6f67..f609c76 100644
--- a/drivers/soc/imx/soc-imx8.c
+++ b/drivers/soc/imx/soc-imx8.c
@@ -217,6 +217,8 @@ static u32 imx8mq_soc_revision(void)
 	{ .compatible = "fsl,imx8qm", .data = &imx8qm_soc_data, },
 	{ .compatible = "fsl,imx8qxp", .data = &imx8qxp_soc_data, },
 	{ .compatible = "fsl,imx8mq", .data = &imx8mq_soc_data, },
+	/* Fixme: this is a hack for big/little xen guest, b0 no need this */
+	{ .compatible = "xen,xenvm", .data = &imx8qm_soc_data, },
 	{ }
 };
 
-- 
1.7.9.5

